samples/meetings-app-icon-badging/csharp/InMeetingNotifications/Bots/InMeetingNotificationsBot.cs

```diff
@@ -0,0 +1,189 @@
+
+                if (turnContext.Activity.Text.Trim() == "SendNotification")
+                {
+                    var meetingPartipantsList = new List<ParticipantDetail>();
+
+                    // Get the list of members that are part of meeting group.
+                    var participants = await TeamsInfo.GetPagedMembersAsync(turnContext);
+
+                    var meetingId = turnContext.Activity.TeamsGetMeetingInfo()?.Id ?? throw new InvalidOperationException("This method is only valid within the scope of a MS Teams Meeting.");
+
+                    foreach (var member in participants.Members)
+                    {
+                        TeamsMeetingParticipant participantDetails = await TeamsInfo.GetMeetingParticipantAsync(turnContext, meetingId, member.AadObjectId, _config["TenandId"]).ConfigureAwait(false);
+
+                        // Select only those members that present when meeting is started.
+                        if (participantDetails.Meeting.InMeeting == true)
+                        {
+                            var meetingParticipant = new ParticipantDetail() { Id = member.Id, Name = member.GivenName };
+                            meetingPartipantsList.Add(meetingParticipant);
+                        }
+                    }
+
+                    var meetingNotificationDetails = new MeetingNotification
+                    {
+                        ParticipantDetails = meetingPartipantsList
+                    };
+
+                    // Send and adaptive card to user to select members for sending targeted notifications.
+                    Attachment adaptiveCardAttachment = GetAdaptiveCardAttachment("SendTargetNotificationCard.json", meetingNotificationDetails);
+                    await turnContext.SendActivityAsync(MessageFactory.Attachment(adaptiveCardAttachment));
+                }
+                else
+                {
+                    await turnContext.SendActivityAsync(MessageFactory.Text("Please type `SendNotification` to send In-meeting notifications."));
+                }
+            }
+            else
+            {
+                await HandleActions(turnContext, cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Method to handle different notification actions in meeting.
+        /// </summary>
+        /// <param name="turnContext">The context object for this turn.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        private async Task HandleActions(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var action = JsonConvert.DeserializeObject<ActionBase>(turnContext.Activity.Value.ToString());
+
+            switch (action.Type)
```

Comment:
Just for my own understanding - 
(1) What other action.type are we planning to support here? 
(2) For the existing scenario (ex: targeted Meeting Notification), we support a different surface type (ex: Meeting Stage). is that scenario in a different bot?

Reply:
Added Stage view as well like nodejs, once testing is done will update accordingly.
 Thanks.